modules/effects/arrow-effects-reactor/src/main/kotlin/arrow/effects/FluxK.kt

```diff
@@ -27,6 +32,14 @@ data class FluxK<A>(val flux: Flux<A>) : FluxKOf<A>, FluxKKindedJ<A> {
   fun <B> flatMap(f: (A) -> FluxKOf<B>): FluxK<B> =
     flux.flatMap { f(it).fix().flux }.k()
 
+  fun <B> bracketCase(use: (A) -> FluxKOf<B>, release: (A, ExitCase<Throwable>) -> FluxKOf<Unit>): FluxK<B> =
+    flatMap { a ->
+      use(a).fix().flux
+        .doOnNext { release(a, ExitCase.Completed) }
```

Comment:
It shouldn't be `doOnNext` because `a` can be used multiple times. It should be on `doOnComplete`.

In any case, check if `doOnTerminate` exists for both cases, and whether they're called when the flux is `cancelled`.

Reply:
I was mistaken about how the callbacks work, sorry. I'm refactored it to use the 3 different callbacks depending on the case now:

```
.doOnError { release(a, ExitCase.Error(it)) }
.doOnCancel { release(a, ExitCase.Cancelled) }
.doOnComplete { release(a, ExitCase.Completed) }
```

I've added a test to ensure the expected behaviour in cancellation.

Reply:
I was mistaken about how the callbacks work, sorry. I'm refactored it to use the 3 different callbacks depending on the case now:

```
.doOnError { release(a, ExitCase.Error(it)) }
.doOnCancel { release(a, ExitCase.Cancelled) }
.doOnComplete { release(a, ExitCase.Completed) }
```

I've added a test to ensure the expected behaviour in cancellation.

Reply:
I was mistaken about how the callbacks work, sorry. I'm refactored it to use the 3 different callbacks depending on the case now:

```
.doOnError { release(a, ExitCase.Error(it)) }
.doOnCancel { release(a, ExitCase.Cancelled) }
.doOnComplete { release(a, ExitCase.Completed) }
```

I've added a test to ensure the expected behaviour in cancellation.

Reply:
I was mistaken about how the callbacks work, sorry. I'm refactored it to use the 3 different callbacks depending on the case now:

```
.doOnError { release(a, ExitCase.Error(it)) }
.doOnCancel { release(a, ExitCase.Cancelled) }
.doOnComplete { release(a, ExitCase.Completed) }
```

I've added a test to ensure the expected behaviour in cancellation.